src/routes/choice-exercises/components/SingleChoiceExercise.tsx

```diff
@@ -81,6 +81,7 @@ const ChoiceExerciseScreen = ({
   }, [results, currentWord])
 
   const onExerciseFinished = async (results: DocumentResult[]): Promise<void> => {
+    const progress = await getExerciseProgress()
     if (disciplineId) {
```

Comment:
Why do we have to check for disciplineId here? Does it make sense to get the progess before saving new progress?